scripts/3_create_v2.sql

```diff
@@ -0,0 +1,59 @@
+create table aws_region
+(
+    id     serial primary key,
+    region varchar not null unique
+);
+
+create table aws_account
+(
+    id      serial primary key,
+    account varchar not null unique
+);
+
+
+create table aws_resource_type
+(
+    id            serial primary key,
+    resource_type varchar not null unique
+);
+
+create table aws_resource
+(
+    id                   bigserial primary key,
+    arn_id               varchar not null, /* nb, this is NOT full arn, it is resource ID like i-5325235 as ARN is redundant and can be reconstructed */
+    aws_account_id       int     not null, /* the account, type and region are immutable, so the lookup tables are referenced as the part of resource, not event */
+    foreign key (aws_account_id) references aws_account (id),
+    aws_region_id        int     not null,
+    foreign key (aws_region_id) references aws_region (id),
+    aws_resource_type_id int     not null,
+    foreign key (aws_resource_type_id) references aws_resource_type (id),
+    meta                 JSONB /* this is not optimal and is not correct as tags can change and changes need tracking */
+);
+
+
+create table aws_public_ip_assignment
+(
+    id              bigserial primary key,
+    not_before      timestamp not null,
+    not_after       timestamp,
+    public_ip       inet,
+    aws_hostname    varchar   not null,
+    aws_resource_id bigint    not null,
+    foreign key (aws_resource_id) references aws_resource (id)
+);
+
+create index idx_public_ip on aws_public_ip_assignment (public_ip);
+
+create index idx_aws_hostname on aws_public_ip_assignment (aws_hostname);
+
+create table aws_private_ip_assignment
+(
+    id              bigserial primary key,
+    not_before      timestamp not null,
+    not_after       timestamp,
+    private_ip      inet,
```

Comment:
is there a reason private_ip is nullable? is there an occasion where we will do an insert into this table without a private IP address?

Reply:
Nope, and this is a great catch. Fixing.